.github/workflows/tests.yml

```diff
@@ -0,0 +1,51 @@
+name: tests-pipelinedp
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  linting:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f requirements.dev.txt ]; then pip install -r requirements.dev.txt; fi
```

Comment:
are all packages of requirements.dev needed?

Reply:
Yes when I checked the tests they are using most requirements